go-controller/pkg/ovn/master.go

```diff
@@ -533,66 +588,60 @@ func (oc *DefaultNetworkController) ensureNodeLogicalNetwork(node *kapi.Node, ho
 		return err
 	}
 
-	// Connect the switch to the router.
-	logicalSwitchPort := nbdb.LogicalSwitchPort{
-		Name:      types.SwitchToRouterPrefix + switchName,
-		Type:      "router",
-		Addresses: []string{"router"},
-		Options:   map[string]string{"router-port": types.RouterToSwitchPrefix + switchName},
-	}
-	sw := nbdb.LogicalSwitch{Name: switchName}
-	err = libovsdbops.CreateOrUpdateLogicalSwitchPortsOnSwitch(oc.nbClient, &sw, &logicalSwitchPort)
-	if err != nil {
-		klog.Errorf("Failed to add logical port %+v to switch %s: %v", logicalSwitchPort, switchName, err)
-		return err
-	}
-
-	err = libovsdbops.AddPortsToPortGroup(oc.nbClient, types.ClusterRtrPortGroupName, logicalSwitchPort.UUID)
-	if err != nil {
-		klog.Errorf(err.Error())
-		return err
-	}
-
 	// Add the switch to the logical switch cache
-	return oc.lsManager.AddSwitch(switchName, logicalSwitch.UUID, hostSubnets)
+	return oc.lsManager.AddSwitch(logicalSwitch.Name, logicalSwitch.UUID, hostSubnets)
```

Comment:
Because this means change the current logic that another addLogicalPort() could continue (it would pass the waitForNodeLogicalSwitch() check) even in this function EnsureJoinLRPIPs() or hostNetworkNamespace nsInfo IP address update fails. Changing the existing code logic is what I am trying to avoid in this PR.

I don't think change that logic is problematic though. So if you feel strongly about it, I can move this into the  createNodeLogicalSwitch() function. Please let me know...

Reply:
Before if `EnsureJoinLRPIPs` failed we would not create the LSP as `EnsureJoinLRPIPs` was being called before that.

Now if `EnsureJoinLRPIPs` fails we are not undoing anything we do in `createNodeLogicalSwitch`

So to me it looks like the best course of action here is to do `EnsureJoinLRPIPs` first, then `createNodeLogicalSwitch` and include `oc.lsManager.AddSwitch` in `createNodeLogicalSwitch`